src/main/java/com/banking/bankservice/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.banking.bankservice.service.impl;
+
+import com.banking.bankservice.dao.UserDao;
+import com.banking.bankservice.exception.EntityNotFoundException;
+import com.banking.bankservice.model.User;
+import com.banking.bankservice.service.UserService;
+import lombok.AllArgsConstructor;
+import org.bson.types.ObjectId;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        return userDao.save(user);
+    }
+
+    @Override
+    public User getById(ObjectId id) {
+        return userDao.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("User " + id + " does not exist"));
+    }
+
+    @Override
+    public User findByPhoneNumber(String phoneNumber) {
+        return userDao.getByPhoneNumber(phoneNumber).orElseThrow(() ->
+                new EntityNotFoundException("No user found"));
```

Comment:
```suggestion
                new EntityNotFoundException("No user found by number " + phoneNumber));
```

Reply:
I didn't really aim to set GitGuardian up, they just emailed me because of the MongoURI string in application.properties :)